includes/html/functions.inc.php

```diff
@@ -1610,92 +1610,57 @@ function get_device_name($device)
+            $state_text = !empty($state_translation[0]['state_descr']) ? $state_translation[0]['state_descr'] : "Unknown";
             $state_label = "label-default";
     }
-    $state = "<span class='label $state_label'>$state_text</span>";
+    $state = "<span class='label $state_label'>$state_text ({$sensor['sensor_current']})</span>";
     return $state;
 }
 
 /**
- * Get state label color
+ * Get sensor label and state color
  */
 function get_sensor_label_color($sensor)
 {
-    $current_label_color = "label-success";
+    $label_style = "label-success";
     if (is_null($sensor)) {
         return "label-unknown";
     }
     if (!is_null($sensor['sensor_limit_warn']) && $sensor['sensor_current'] > $sensor['sensor_limit_warn']) {
-        $current_label_color = "label-warning";
+        $label_style = "label-warning";
     }
     if (!is_null($sensor['sensor_limit_low_warn']) && $sensor['sensor_current'] < $sensor['sensor_limit_low_warn']) {
-        $current_label_color = "label-warning";
+        $label_style = "label-warning";
     }
     if (!is_null($sensor['sensor_limit']) && $sensor['sensor_current'] > $sensor['sensor_limit']) {
-        $current_label_color = "label-danger";
+        $label_style = "label-danger";
     }
     if (!is_null($sensor['sensor_limit_low']) && $sensor['sensor_current'] < $sensor['sensor_limit_low']) {
-        $current_label_color = "label-danger";
+        $label_style = "label-danger";
     }
 
-    return $current_label_color;
-}
-
-/**
- * Get the unit for the sensor class given as parameter
- * @param $class
- * @return string The unit
- */
-function get_unit_for_sensor_class($class)
-{
-    $units_by_classes = array(
-        'ber'                  => '',
-        'charge'               => '%',
-        'chromatic_dispersion' => 'ps/nm',
-        'cooling'              => 'W',
-        'count'                => '',
-        'current'              => 'A',
-        'dbm'                  => 'dBm',
-        'delay'                => 's',
-        'eer'                  => '',
-        'fanspeed'             => 'rpm',
-        'frequency'            => 'Hz',
-        'humidity'             => '%',
-        'load'                 => '%',
-        'power'                => 'W',
-        'power_consumed'       => 'kWh',
-        'power_factor'         => '',
-        'pressure'             => 'kPa',
-        'quality_factor'       => 'dB',
-        'signal'               => 'dBm',
-        'snr'                  => 'dB',
-        'state'                => '',
-        'temperature'          => '&deg;C',
-        'voltage'              => 'V',
-        'waterflow'            => 'l/m',
-    );
-
-    if (!array_key_exists($class, $units_by_classes)) {
-        return '';
-    }
+    $unit = __('sensors.' . $sensor['sensor_class'] . '.unit');
 
-    return $units_by_classes[$class];
+    $sensor_current = "<span class='label $label_style'>".trim(format_si($sensor['sensor_current']).$unit)."</span>";
```

Comment:
You could probably inline this variable here and just return the string

Reply:
Updated that as requested for both the get_state_label() and get_sensor_label_color() functions!